core/templates/pages/exploration-editor-page/editor-tab/templates/modal-templates/add-hint-modal.component.ts

```diff
@@ -73,7 +73,19 @@ export class AddHintModalComponent
   }
 
   isHintLengthExceeded(tmpHint: string): boolean {
-    return Boolean(tmpHint.length > 500);
+    // The variable charCount stores the number of remaining characters after
+    // removing the html tags from tmpHint.
+    let doc = new DOMParser;
+    let charCount = 0;
+    if (tmpHint.length) {
+      let dom = doc.parseFromString(tmpHint, 'text/html');
+      let newLineRegex = /(\n)(\s\n)*/g;
```

Comment:
Looks like this is a constant, if so should we define it as a constant (see how constants are used in other components)?

Reply:
Done